tt_metal/hw/inc/dev_msgs.h

```diff
@@ -254,6 +254,27 @@ struct profiler_msg_t {
     uint32_t buffer[PROFILER_RISC_COUNT][kernel_profiler::PROFILER_L1_VECTOR_SIZE];
 };
 
+struct addressable_core_t {
+    volatile uint8_t x, y;
+    volatile AddressableCoreType type;
+};
+
+// TODO: This can move into the hal eventually, currently sized for WH.
+constexpr static std::uint32_t MAX_NON_WORKER_CORES = 36 + 1 + 16;
```

Comment:
dram + pcie + eth? for BH this is 24 + 1 + 14. Maybe open up an issue for Hal to uplift this

Reply:
Filed an issue, I see there's some other TODOs in the same file to uplift w/ hal so group them together: https://github.com/tenstorrent/tt-metal/issues/12746

Yeah it's dram + pcie + eth, was checking the .yamls and found WH to be the highest with 36 + 1 + 16 so sized it for that